pkgs/build-support/rust/carnix.nix

```diff
@@ -0,0 +1,536 @@
+      crateName = "log";
+      version = "0.3.8";
+      sha256 = "1c43z4z85sxrsgir4s1hi84558ab5ic7jrn5qgmsiqcv90vvn006";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    lru_cache_0_1_1_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "lru-cache";
+      version = "0.1.1";
+      sha256 = "1hl6kii1g54sq649gnscv858mmw7a02xj081l4vcgvrswdi2z8fw";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    memchr_1_0_2_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "memchr";
+      version = "1.0.2";
+      sha256 = "0dfb8ifl9nrc9kzgd5z91q6qg87sh285q1ih7xgrsglmqfav9lg7";
+      libName = "memchr";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    nom_3_2_1_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "nom";
+      version = "3.2.1";
+      sha256 = "1vcllxrz9hdw6j25kn020ka3psz1vkaqh1hm3yfak2240zrxgi07";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    num_traits_0_1_40_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "num-traits";
+      version = "0.1.40";
+      sha256 = "1fr8ghp4i97q3agki54i0hpmqxv3s65i2mqd1pinc7w7arc3fplw";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    pkg_config_0_3_9_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "pkg-config";
+      version = "0.3.9";
+      sha256 = "06k8fxgrsrxj8mjpjcq1n7mn2p1shpxif4zg9y5h09c7vy20s146";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    quote_0_3_15_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "quote";
+      version = "0.3.15";
+      sha256 = "09il61jv4kd1360spaj46qwyl21fv1qz18fsv2jra8wdnlgl5jsg";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    rand_0_3_18_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "rand";
+      version = "0.3.18";
+      sha256 = "15d7c3myn968dzjs0a2pgv58hzdavxnq6swgj032lw2v966ir4xv";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    redox_syscall_0_1_31_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "redox_syscall";
+      version = "0.1.31";
+      sha256 = "0kipd9qslzin4fgj4jrxv6yz5l3l71gnbd7fq1jhk2j7f2sq33j4";
+      libName = "syscall";
+      inherit dependencies buildDependencies features release verbose;
+    };
+    redox_termios_0_1_1_ = { dependencies?[], buildDependencies?[], features?[] }: mkRustCrate {
+      crateName = "redox_termios";
+      version = "0.1.1";
+      sha256 = "04s6yyzjca552hdaqlvqhp3vw0zqbc304md5czyd3axh56iry8wh";
+      libPath = "src/lib.rs";
+      libName = "redox_termios";
```

Comment:
If `libName` equals `crateName` is it still required to set `libName`?

Reply:
This is indeed done in the latest version, I've just published it on crates.io, will update soon.